node/builder_chain.go

```diff
@@ -85,11 +85,18 @@ var ChainNode = Options(
 	Override(new(dtypes.ChainBlockService), modules.ChainBlockService), // todo: unused
 
 	// Consensus: Chain sync
-
 	// We don't want the SyncManagerCtor to be used as an fx constructor, but rather as a value.
 	// It will be called implicitly by the Syncer constructor.
 	Override(new(chain.SyncManagerCtor), func() chain.SyncManagerCtor { return chain.NewSyncManager }),
-	Override(new(*chain.Syncer), modules.NewSyncer),
+
+	ApplyIf(isNotFollower,
+		Override(new(*chain.Syncer), modules.NewSyncer),
+	),
+
+	ApplyIf(isFollower,
+		Override(new(*chain.Syncer), modules.NewSyncerFollower),
+	),
```

Comment:
You don't want the syncer in Follower mode at all.

To make node construction work without it you will likely need to move some more things into `not follower` block, like the Hello protocol handler. Also likely you'll need to add `optional:"true"` tag to Syncer fields in `node/impl` in API structs which have fx.In

Then instead of running the real syncer, you'd write a new module, which takes `*store.ChainStore` to call PutTipSet on with new heads, and `api.Gateway` (which you'll get from cmd/lotus/daemon options) to call ChainNotify on which will get you a channel with head changes